tests/fixtures/plugin/build.atpkg

```diff
@@ -0,0 +1,29 @@
+;; Copyright (c) 2016 Anarchy Tools Contributors.
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;   http:;;www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+
+(package
+  :name "plugin"
+  
+  :tasks {
+    :a {:tool "nop" }
+    :default {
+      :tool "echo.plugin"
+      :key "value"
+      :test "${test_substitution}"
+      :userpath "\${ATBUILD_USER_PATH}"
```

Comment:
This isn't a toolpath, it's an arbitrary key that tests shell variable expansion.  The particular shell variable we expand happens to be called `ATBUILD_USER_PATH` which is documented [here](http://anarchytools.org/docs/shell.html).